src/server/pps/server/api_server.go

```diff
@@ -2362,6 +2462,64 @@ func (a *apiServer) GarbageCollect(ctx context.Context, request *pps.GarbageColl
 	return &pps.GarbageCollectResponse{}, nil
 }
 
+func (a *apiServer) ActivateAuth(ctx context.Context, req *pps.ActivateAuthRequest) (resp *pps.ActivateAuthResponse, retErr error) {
+	func() { a.Log(req, nil, nil, 0) }()
+	defer func(start time.Time) { a.Log(req, resp, retErr, time.Since(start)) }(time.Now())
+	pachClient := a.getPachClient().WithCtx(ctx)
+	ctx = pachClient.Ctx() // pachClient will propagate auth info
+
+	// TODO: block creating of pipelines until this is done, so that ListPipelines
+	// is exhaustive
+	var pipelines []*pps.PipelineInfo
+	if err := a.sudo(pachClient, func(superUserClient *client.APIClient) error {
+		var err error
+		pipelines, err = superUserClient.ListPipeline()
+		if err != nil {
+			return fmt.Errorf("cannot get list of pipelines to update: %v", err)
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	var eg errgroup.Group
+	for _, pipeline := range pipelines {
```

Comment:
It's possible that this could race with pipelines being created right? Probably not that big a deal since auth activation is a very rare occurrence. But how would things fail if it did race and a pipeline missed auth getting activated?

Reply:
Ah, so the idea is that when users call `auth.Activate()`, it "activates" auth with one administrator: the magic user/pps user (it's impossible to log in as the magic user, but auth.isActivate() will return true). At that point, any calls that do auth checks will fail (repos have no ACLs, and there is only one cluster administrator), but pps will still be able to do things. If anything afterwards fails, you can just keep retrying `auth.Activate()` until `pps.ActivateAuth()` succeeds, at which point `auth.Activate()` makes the caller a cluster administrator and the cluster is fully auth-on